### ISSUE TITLE
Add Typescript

### ISSUE THREAD

**Issue by cloetensbrecht** [BasicPrimitives/react#74]

**Comment by BasicPrimitives** [comment]
Yes, supporting TypeScript is a good idea. However, we are currently considering a slightly different improvement, which is similar to the idea of rewriting the entire library into TypeScript.

**Comment by BasicPrimitives** [comment]
I just cannot promise typescript in the near future.

**Comment by amochkin** [comment]
Typescript support is a blocker for us, and we are considering OEM. No need to rewrite whole library, type definitions are enough for that.

**Comment by BasicPrimitives** [comment]
Hi Al, Is the following folder what you need? 
https://github.com/BasicPrimitives/angular/tree/main/projects/ngx-basic-primitives/src/lib/configs
